File: objc/Project1/Project1/Hater.m
//
//  Hater.m
//  Project1
//
//  Created by Jinwoo Kim on 2/22/21.
//

#import "Hater.h"

@implementation Hater

- (instancetype)init {
    self = [super init];
    
    if (self) {
        self.hating = NO;
    }
    
    return self;
}

- (void)hadABadDay {
    self.hating = YES;
}

- (void)hadAGoodDay {
    self.hating = NO;
}

@end

File: swift/Project1/Project1/Hater.swift
//
//  Hater.swift
//  Project1
//
//  Created by Jinwoo Kim on 2/22/21.
//

import Foundation

struct Hater {
    var hating = false
    
    mutating func hadABadDay() {
        hating = true
    }
    
    mutating func hadAGoodDay() {
        hating = false
    }
}
